lib/models/questionnaire.dart

```diff
@@ -145,6 +131,11 @@ class Questionnaire {
     }
   }
 
+  Future<void> updateLanguage() async {
+    _entries.forEach((QuestionnaireEntry entry) {
+      entry.question = questionCatalogF[entry.question.runtimeId];
+    });
+  }
```

Comment:
As discussed last time, the runtimeId is not a future safe way to get the index.

Change this function to:


```dart
/// This should only be called when the question catalog's strings changed.
/// Do not call this method when the question catalog strucutre changed.
void updateQuestionCatalogLanguage(QuestionCatalog questionCatalog) {
   assert(_questionCatalog.length == questionCatalog.length, "Question catalogs are different while they should be of the same strucutre.");
   _questionCatalog = questionCatalog;
   for (var i = 0; i < _entries.length; i++) {
      _entries[i] = _entries[i].copyWith(
         // find the question definition with the same id and replace it with the previous one
         question: _questionCatalog.firstWhere((q) => q == _entries[i].questionDefinition),
      );
   }
}
```

Reply:
I use the RuntimeId because of the _entries[i].questionDefinition will not be equal since "description" and "question" will be different (the text was translated). Because you change the QuestionCatalog to be iterable and include an index then the runtime is the same since  At this moment of the application, the QuestionCatalog has the question in the same order in English and German. Then I thought there was no other way to find the question translation.